From: Debian Rust Maintainers <pkg-rust-maintainers@alioth-lists.debian.net>
Date: Thu, 14 Jul 2022 13:17:39 +0200
Subject: Change i686 to match Debian i386 baseline

see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=973414 , might need to be
adapted to reduce the baseline again

Forwarded: not-needed

===================================================================
---
 compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs | 2 +-
 tests/ui/abi/homogenous-floats-target-feature-mixup.rs           | 3 ++-
 tests/ui/sse2.rs                                                 | 2 +-
 3 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs b/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs
index 1d4916c..aca4928 100644
--- a/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs
+++ b/compiler/rustc_target/src/spec/targets/i686_unknown_linux_gnu.rs
@@ -2,7 +2,7 @@ use crate::spec::{base, Cc, LinkerFlavor, Lld, SanitizerSet, StackProbeType, Tar
 
 pub fn target() -> Target {
     let mut base = base::linux_gnu::opts();
-    base.cpu = "pentium4".into();
+    base.cpu = "pentiumpro".into();
     base.max_atomic_width = Some(64);
     base.supported_sanitizers = SanitizerSet::ADDRESS;
     base.add_pre_link_args(LinkerFlavor::Gnu(Cc::Yes, Lld::No), &["-m32"]);
diff --git a/tests/ui/abi/homogenous-floats-target-feature-mixup.rs b/tests/ui/abi/homogenous-floats-target-feature-mixup.rs
index 3a8540a..bc4fea5 100644
--- a/tests/ui/abi/homogenous-floats-target-feature-mixup.rs
+++ b/tests/ui/abi/homogenous-floats-target-feature-mixup.rs
@@ -24,7 +24,8 @@ fn main() {
     match std::env::var("TARGET") {
         Ok(s) => {
             // Skip this tests on i586-unknown-linux-gnu where sse2 is disabled
-            if s.contains("i586") {
+            // Debian: our i686 doesn't have SSE 2..
+            if s.contains("i586") || s.contains("i686") {
                 return
             }
         }
diff --git a/tests/ui/sse2.rs b/tests/ui/sse2.rs
index fa6d797..8822846 100644
--- a/tests/ui/sse2.rs
+++ b/tests/ui/sse2.rs
@@ -15,7 +15,7 @@ fn main() {
         }
         Err(_) => return,
     }
-    if cfg!(any(target_arch = "x86", target_arch = "x86_64")) {
+    if cfg!(any(target_arch = "x86_64")) {
         assert!(cfg!(target_feature = "sse2"),
                 "SSE2 was not detected as available on an x86 platform");
     }
